worlds/banjo_tooie/BTClient.py

```diff
@@ -0,0 +1,1248 @@
+        message = "Attempting to close this window again will forcibly close it."
+        def cleanup(messagebox):
+            self._messagebox = None
+        if self._messagebox and self._messagebox.title == title:
+            return False
+        if program and program.poll() == None:
+            self.gui_error(title, message)
+            self._messagebox.bind(on_dismiss=cleanup)
+            return True
+        return False
+
+    def on_package(self, cmd, args):
+        if cmd == "Connected":
+            self.tracker = BanjoTooieItemTracker(self)
+            self.slot_data = args.get("slot_data", None)
+            if version != self.slot_data["custom_bt_data"]["version"]:
+                logger.error("Your Banjo-Tooie AP does not match with the generated world.")
+                logger.error("Your version: "+version+" | Generated version: "+self.slot_data["custom_bt_data"]["version"])
+                # self.event_invalid_game()
+                raise Exception("Your Banjo-Tooie AP does not match with the generated world.\n" +
+                                "Your version: "+version+" | Generated version: "+self.slot_data["custom_bt_data"]["version"])
+            self.deathlink_enabled = bool(self.slot_data["options"]["death_link"])
+            self.taglink_enabled = bool(self.slot_data["options"]["tag_link"])
+            self.n64_sync_task = asyncio.create_task(n64_sync_task(self), name="N64 Sync")
+        elif cmd == "ReceivedItems":
+            self.tracker.refresh_items()
+            if self.startup == False:
+                for item in args["items"]:
+                    player = ""
+                    item_name = ""
+                    for (i, name) in self.player_names.items():
+                        if i == item.player:
+                            player = name
+                            break
+                    for (name, i) in bt_itm_name_to_id.items():
+                        if item.item == i:
+                            item_name = name
+                            break
+                    logger.info(player + " sent " + item_name)
+                logger.info("The above items will be sent when Banjo-Tooie is loaded.")
+                self.startup = True
+        if isinstance(args, dict) and isinstance(args.get("data", {}), dict):
+            source_name = args.get("data", {}).get("source", None)
+            if not hasattr(self, "instance_id"):
+                self.instance_id = time.time()
+            if "TagLink" in self.tags and source_name != self.instance_id and "TagLink" in args.get("tags", []):
+                if not hasattr(self, "pending_tag_link"):
+                    self.pending_tag_link = False
+                else:
+                    self.pending_tag_link = True
+
+    def on_print_json(self, args: dict):
+        if self.ui:
+            self.ui.print_json(copy.deepcopy(args["data"]))
+            relevant = args.get("type", None) in {"ItemSend"}
```

Comment:
```suggestion
            relevant = args.get("type") == "ItemSend"
```
Slightly overcomplicated.